pkg/image/validation/registry_test.go

```diff
@@ -0,0 +1,238 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+)
+
+func TestValidateEmbeddedArtefactRegistry(t *testing.T) {
+	tests := map[string]struct {
+		Registry               image.EmbeddedArtifactRegistry
+		ExpectedFailedMessages []string
+	}{
+		`no registry`: {
+			Registry: image.EmbeddedArtifactRegistry{},
+		},
+		`full valid example`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				ContainerImages: []image.ContainerImage{
+					{
+						Name:           "foo",
+						SupplyChainKey: "key",
+					},
+				},
+				HelmCharts: []image.HelmChart{
+					{
+						Name:    "bar",
+						RepoURL: "http://bar.com",
+						Version: "3.14",
+					},
+				},
+			},
+		},
+		`failures in both sections`: {
+			Registry: image.EmbeddedArtifactRegistry{
+				ContainerImages: []image.ContainerImage{
+					{
+						Name: "", // trips the missing name validation
+					},
+				},
+				HelmCharts: []image.HelmChart{
+					{
+						Name:    "", // trips the missing name validation
+						RepoURL: "http://doesntmatter.com",
+						Version: "31",
+					},
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"The 'name' field is required for each entry in 'images'.",
+				"The 'name' field is required for each entry in 'charts'.",
+			},
+		},
+	}
```

Comment:
Another potential nit, I think this coverage is fine, but I think I'd also like to see individual failure cases.

Reply:
The individual failure cases are covered in the respective tests for validating container images and helm charts. This test simply shows that both of those are called from this runner function.